pkgs/applications/networking/instant-messengers/discord/default.nix

```diff
@@ -61,8 +61,11 @@ let
       ++ lib.optionals (branch == "ptb") [ "aarch64-darwin" ];
   };
   package = if stdenv.isLinux then ./linux.nix else ./darwin.nix;
+
+  openasar = if withOpenASAR then callPackage ./openasar.nix { } else null;
+
   packages = (builtins.mapAttrs
-    (_: value: callPackage package (value // { inherit src version; meta = meta // { mainProgram = value.binaryName; }; }))
+    (_: value: callPackage package (value // { inherit src version openasar; meta = meta // { mainProgram = value.binaryName; }; }))
```

Comment:
BTW, we shouldn't really be using `callPackage` inside the derivations itself. I think it is fine for now because the derivation already works like this, but I think this derivation should be refactored to extract its packages from the inputs.